src/core/reactor.cc

```diff
@@ -923,9 +966,14 @@ reactor::task_queue::register_stats() {
 }
 
 void
-reactor::task_queue::rename(sstring new_name) {
+reactor::task_queue::rename(sstring new_name, sstring new_shortname) {
     if (_name != new_name) {
         _name = new_name;
+        if (new_shortname.empty()) {
+            _shortname = shorten_name(_name, shortname_size);
+        } else {
+            _shortname = new_shortname;
```

Comment:
Should it be in sync with the constructor and do `format("{:>{}}", new_shortname, shortname_size)` thing as well?